monai/main.py

```diff
@@ -0,0 +1,604 @@
+import os
+import argparse
+from datetime import datetime
+from loguru import logger
+
+import numpy as np
+import wandb
+import torch
+import pytorch_lightning as pl
+import torch.nn.functional as F
+import matplotlib.pyplot as plt
+
+from utils import precision_score, recall_score, dice_score, plot_slices, PolyLRScheduler
+from losses import SoftDiceLoss
+from transforms import train_transforms, val_transforms, test_transforms
+
+from monai.utils import set_determinism
+from monai.inferers import sliding_window_inference
+from monai.networks.nets import UNet, DynUNet, BasicUNet, UNETR
+from monai.data import (DataLoader, Dataset, CacheDataset, load_decathlon_datalist, decollate_batch)
+from monai.transforms import (Compose, EnsureType, EnsureTyped, Invertd, SaveImaged, SaveImage)
+
+
+# create a "model"-agnostic class with PL to use different models
+class Model(pl.LightningModule):
+    def __init__(self, args, data_root, fold_num, net, loss_function, optimizer_class, 
+                 exp_id=None, results_path=None):
+        super().__init__()
+        self.args = args
+        self.save_hyperparameters(ignore=['net'])
+
+        # if self.args.unet_depth == 3:
+        #     from models import ModifiedUNet3DEncoder, ModifiedUNet3DDecoder   # this is 3-level UNet
+        #     logger.info("Using UNet with Depth = 3! ")
+        # else:
+        #     from models_original import ModifiedUNet3DEncoder, ModifiedUNet3DDecoder
+        #     logger.info("Using UNet with Depth = 4! ")
+
+        self.root = data_root
+        self.fold_num = fold_num
+        self.net = net
+        # self.load_pretrained = load_pretrained
+        self.lr = args.learning_rate
+        self.loss_function = loss_function
+        self.optimizer_class = optimizer_class
+        self.save_exp_id = exp_id
+        self.results_path = results_path
+
+        self.best_val_dice, self.best_val_epoch = 0, 0
+
+        # define cropping and padding dimensions
+        # NOTE: taken from nnUNet_plans.json
+        self.voxel_cropping_size = (64, 128, 128)   # (80, 192, 160) 
```

Comment:
```suggestion
        self.voxel_cropping_size = (64, 128, 128)   # (80, 192, 160) from nnUnet
```

Reply:
thank you for suggesting this - has been fixed in 72ac6fe49e362dddc1082c9426416f32d6c994f1